### ISSUE TITLE
[BUG] error: dereferencing type-punned pointer will break strict-aliasing rules

### ISSUE THREAD

**Issue by tbnobody** [bertmelis/espMqttClient#80]
**Describe the bug**
I was using the lib in version 1.3.3 and ugpraded to 1.4.0.

I am getting the following error message:
```
.pio/libdeps/generic/espMqttClient/src/MqttClient.cpp: In member function 'void MqttClient::_checkTimeout()':
.pio/libdeps/generic/espMqttClient/src/MqttClient.cpp:465:22: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
     if (millis() - *((uint32_t*)&(it.get()->token)) > _timeout) {  // NOLINT(readability/casting)
```

I am buildin my source with 
`-Wall -Wextra -Werror`

**Comment by bertmelis** [comment]
Yeah... I'm aware of that, there's a to-do in the code.

On esp you can safely ignore the warning but if you enforce warnings as errors with compiler flags you cannot obviously.

There are several options to solve this. I'm not sure which one to pick.

**Comment by bertmelis** [comment]
May I ask. How do you achieve -Werror? There are warnings in the Arduino framework itself. How do you compile that?

**Comment by tbnobody** [comment]
It's a ESP32. 
```
framework = arduino
platform = espressif32@>=6.0.1
board = esp32dev
```

Compiles in platformio without any problems

**Comment by bertmelis** [comment]
Should be fixed with #81
